test/instrumentation/async-hooks.js

```diff
@@ -125,12 +125,20 @@ test('sync/async tracking', function (t) {
   var trans = agent.startTransaction()
   t.strictEqual(trans.sync, true)
 
-  var span = agent.startSpan()
-  t.strictEqual(span.sync, true)
+  var span1 = agent.startSpan()
```

Comment:
I see in this test we're explicitly starting and stopping `span2` in the same execution/async context, and then making sure its `sync` value remains `true`.  I also see we're testing `span1`'s `sync` value _inside_ the `setImmediate` async context to make sure it's been identified as an asynchronous span.